### DISCUSSION TITLE
Stability of VITS

### DISCUSSION THREAD

**Discussion by shigabeev** [coqui-ai/TTS#2121] [General Q&A]
Hi, I'm training VITS with regular config https://github.com/coqui-ai/TTS/tree/dev/recipes/multilingual/vits_tts, but on LibriTTS instead of MAILabs and `num_layers_text_encoder = 10` .

In general, produced speech sounds good. But the speech quality reduces significantly when synthesizing sentences longer than 300 characters. Mainly due to duration predictor. **How to make VITS stable on longer sequences?**

Here's how it regularly sounds:

https://user-images.githubusercontent.com/9149327/199984062-8c08abb2-af6d-4379-8ede-abee0eee90fd.mp4

Text:
```
Nineteen-year-old Alice returns to the magical world from her childhood adventure, where she reunites with her old friends and learns of her true destiny.
```

Now, now longer audio sounds:

https://user-images.githubusercontent.com/9149327/199984153-e2b2b36c-9d0e-4f06-9d8a-665362feec62.mp4

text:
```
With that money, we have researched our target market and technologies, documented our product specification, designed the system architecture, built our core development team, completed our prototype, and protected our development with several significant patent applications, many of which have already been allowed and issued, and even received our first customers, who enjoy our product and looking forward to see the launch of it!.
```

Fun fact. Model, trained on VCTK produces clean sound on the 2X longer example:

https://user-images.githubusercontent.com/9149327/199987414-a2a0d8eb-95c9-446f-bfee-3d031113561a.mp4

The most relevant topic here
https://github.com/coqui-ai/TTS/issues/2078,
but there, the author uses the Tacotron2, which is of very different architecture, so the advice from there isn't applicable. 

I noticed that models located in the model zoo of the repository are stable enough on longer sequences. How can I make my models invariant to the length of the sentence too? Of course, I can split input sequences into sentences (which I do) and even further on commas and connector words, but I'm sure there are better solutions than this.

My config file:
[config.txt](https://github.com/coqui-ai/TTS/files/9938937/config.txt)

Thanks in advance!

**Comment by yyhsmdqj** [comment]
已收到，谢谢您！
